pkg/generate/app/sourcelookup.go

```diff
@@ -378,7 +380,7 @@ func StrategyAndSourceForRepository(repo *SourceRepository, image *ImageRef) (*B
 		Binary: repo.binary,
 	}
 
-	if repo.Info() != nil && repo.Info().Dockerfile != nil {
+	if (repo.ignoreRepository || repo.dockerfile) && repo.Info() != nil && repo.Info().Dockerfile != nil {
```

Comment:
It is coming back to fix the bug that was introduced. The idea is that I'm reverting to how it worked before, and now adding a new bool that is only used by new-build.